component/src/main/java/org/wso2/carbon/identity/authenticator/reddit/RedditAuthenticator.java

```diff
@@ -0,0 +1,296 @@
+        try {
+            oAuthResponse = oAuthClient.accessToken(accessRequest);
+        } catch (OAuthSystemException | OAuthProblemException e) {
+            throw new AuthenticationFailedException("Exception while requesting access token", e);
+        }
+        return oAuthResponse;
+    }
+
+    /**
+     * Request user claims from user info endpoint.
+     *
+     * @param url         User info endpoint.
+     * @param accessToken Access token.
+     * @return Response string.
+     * @throws IOException
+     */
+    @Override
+    protected String sendRequest(String url, String accessToken)
+            throws IOException {
+
+        if (log.isDebugEnabled()) {
+            log.debug("Claim URL: " + url);
+        }
+        if (url == null) {
+            return StringUtils.EMPTY;
+        }
+        URL obj = new URL(url);
+        HttpURLConnection urlConnection = (HttpURLConnection) obj.openConnection();
+        urlConnection.setRequestMethod(RedditAuthenticatorConstants.HTTP_GET);
+        urlConnection.setRequestProperty(RedditAuthenticatorConstants.AUTHORIZATION,
+                RedditAuthenticatorConstants.AUTHORIZATION_BEARER + accessToken);
+        urlConnection.setRequestProperty(RedditAuthenticatorConstants.CONTENT_TYPE,
+                RedditAuthenticatorConstants.CONTENT_TYPE_JSON);
+        urlConnection.setRequestProperty(RedditAuthenticatorConstants.REQUEST_HOST_HEADER,
+                RedditAuthenticatorConstants.REDDIT_HOST);
+        urlConnection.setRequestProperty(RedditAuthenticatorConstants.REDDIT_USER_AGENT,
+                RedditAuthenticatorConstants.USER_AGENT);
+        BufferedReader reader = new BufferedReader(new InputStreamReader(urlConnection.getInputStream()));
+        StringBuilder builder = new StringBuilder();
+        String inputLine = reader.readLine();
+        while (inputLine != null) {
+            builder.append(inputLine).append("\n");
+            inputLine = reader.readLine();
+        }
+        reader.close();
+        if (log.isDebugEnabled() && IdentityUtil.isTokenLoggable(IdentityConstants.IdentityTokens.USER_ID_TOKEN)) {
+            log.debug("response: " + builder.toString());
+        }
+        return builder.toString();
+    }
+
+    /**
+     * Get Configuration Properties
+     */
+    @Override
+    public List<Property> getConfigurationProperties() {
+
+        List<Property> configProperties = new ArrayList<>();
+
+        Property clientId = new Property();
+        clientId.setName(OIDCAuthenticatorConstants.CLIENT_ID);
+        clientId.setDisplayName("Client Id");
+        clientId.setRequired(true);
+        clientId.setDescription("Enter Reddit  client identifier value");
+        clientId.setDisplayOrder(1);
+        configProperties.add(clientId);
+
+        Property clientSecret = new Property();
+        clientSecret.setName(OIDCAuthenticatorConstants.CLIENT_SECRET);
+        clientSecret.setDisplayName("Client Secret");
```

Comment:
Fixed in https://github.com/wso2-extensions/identity-outbound-auth-reddit/pull/3